src/fsharp/FSharp.Core/prim-types.fsi

```diff
@@ -1336,8 +1339,38 @@ namespace Microsoft.FSharp.Core
             //[<CompilerMessage("This function is for use by compiled F# code and should not be used directly", 1204, IsHidden=true)>]
             val inline SetArray4D : target:'T[,,,] -> index1:int -> index2:int -> index3:int -> index4:int -> value:'T -> unit  
 
+        module internal Reflection =
+            val internal tupleNames : string []
+            val internal isTupleType : Type -> bool
+            val internal tryFindSourceConstructFlagsOfType : Type * byref<SourceConstructFlags> -> bool
+            val internal fieldPropsOfRecordType : Type * System.Reflection.BindingFlags -> System.Reflection.PropertyInfo[]
+            val internal isRecordType : Type * System.Reflection.BindingFlags -> bool
+            val internal getUnionTypeTagNameMap : Type * System.Reflection.BindingFlags -> (int*string)[]
+            val internal fieldsPropsOfUnionCase : Type * int* System.Reflection.BindingFlags -> System.Reflection.PropertyInfo[]
+            val internal isUnionType : Type * System.Reflection.BindingFlags -> bool
+
         /// <summary>The F# compiler emits calls to some of the functions in this module as part of the compiled form of some language constructs</summary>
         module HashCompare =
+            [<AbstractClass; Sealed>]
+            type FSharpEqualityComparer_ER<'T> =
```

Comment:
These are now public APIs.  We need to be sure the names are what we would want.

Reply:
Way past my pay-grade! @cartermp ?

Reply:
Hmm.. Some thoughts here though; maybe, to mimic `EqualityComparer.Default`, `FSharpEqualityComparer_PER.EqualityComparer` should become `FSharpEqualityCompare.Default` (and just leave `FSharpEqualityCompare_ER`... As I think that one is the lesser used...) But then is it right to leave it in it's current namespace? `EqualityComparer` lives in `System.Collections.Generic` so should it be in `FSharp.Collections`? Or should it remain 'public'-lite, i.e. not really for use by anyone except the compiler...